transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java

```diff
@@ -119,9 +119,9 @@ public boolean processMessage(Object msg) {
                 ByteBuf buf = packet.content();
                 return add0(buf, buf.readerIndex(), buf.readableBytes(), packet.recipient());
```

Comment:
I think even here it makes sense to pass `null` if connected... since the address will be ignored IIUC

Reply:
I don't think this is true... even on connected channels you can send to other addresses IIUC.

Reply:
I'm not sure how the syscalls behave but java `DatagramChannel` does not allow it, and the NIO impl will ignore other addresses if connected: https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/socket/nio/NioDatagramChannel.java#L291

Reply:
I am a bit confused... in the code you linked we will always pass the remoteAddress of the `DatagramPacket` if there is one even if we are connected. So it is inline with what we have here now... Also I tested what happens if I set something invalid in native code for the addr and it failed the syscall. So it seems like it is not ignored (like I suspected).

Reply:
Oops :) you are right sorry I misread the `remoteAddress` there as being the channel's address rather than the packet's address ... _but_ this won't actually work if you attempt it. So the NIO impl is kind of broken in this regard
```
Caused by: java.lang.IllegalArgumentException: Connected address not equal to target address
	at sun.nio.ch.DatagramChannelImpl.send(DatagramChannelImpl.java:448)
	at io.netty.channel.socket.nio.NioDatagramChannel.doWriteMessage(NioDatagramChannel.java:292)
```
but I think you're right that the kernel allows it so I guess fine to leave as-is